src/Uno.UI/UI/Xaml/Controls/Popup/Popup.Android.cs

```diff
@@ -36,6 +41,17 @@ internal Popup(bool useNativePopup) : this()
 			OnIsLightDismissEnabledChanged(false, true);
 
 			_popupWindow.DismissEvent += OnPopupDismissed;
+			_disposable.Disposable = Disposable.Create(() => _popupWindow.DismissEvent -= OnPopupDismissed);
+		}
+
+		private void ReinitializePopupWindowIfContextChanged()
```

Comment:
add a comment explaining why this method is needed

Reply:
It would also be interesting to add a UI test flagged as manual, so we can validate it in the future.